### DISCUSSION TITLE
Where is the json schema of vscode settings.json?

### DISCUSSION THREAD

**Discussion by jcrben** [microsoft/vscode-discussions#507] [Extension Development QnA] [answered]
Anyone up for answering this question? https://stackoverflow.com/questions/66838431/where-is-the-json-schema-of-vscode-settings-json

I spelunked through the codebase a bit and noticed https://github.com/microsoft/vscode/blob/274f338db414b52130053fe5a4118913f5275532/src/vs/workbench/services/configuration/common/configuration.ts#L18-L25

This is not directly related to extension development but could be useful to someone and no answer was received from the "recommended" channel.

I want this because I have a file of settings which my team and I are sharing and it's not getting intellisense since it's not in the usual place.

**Comment by usernamehw** [comment, marked as answer]
Maybe it's dynamically generated? If you just want autocomplete:

```js
"json.schemas": [
	{
		"fileMatch": [
			"settings.json"
		],
		"url": "vscode://schemas/settings/user"
	}
]
```

If you want its contents from extension:

```ts
const document = await workspace.openTextDocument(Uri.parse('vscode://schemas/settings/user'));
const documentText = document.getText();
```

It would be something like: https://github.com/wraith13/vscode-schemas/blob/master/en/latest/schemas/settings/user.json
